lib/ansible/module_utils/facts.py

```diff
@@ -2498,6 +2498,27 @@ def get_virtual_facts(self):
                 if 'VirtualBox' in line:
                     self.facts['virtualization_type'] = 'virtualbox'
                     self.facts['virtualization_role'] = 'guest'
+        # Detect domaining on Sparc hardware
+        if os.path.exists("/usr/sbin/virtinfo"):
```

Comment:
I took that test from the above section 15 lines up testing for `/usr/bin/zonename`. I can assure you that you cannot safely move stuff around in Solaris without breaking the packaging system and workflows.